official/keras_application_models/model_callbacks.py

```diff
@@ -0,0 +1,168 @@
+      self._logger.log_metric(
+          "current_examples_per_sec", current_examples_per_sec,
+          global_step=self._global_step)
+      self._last_recorded_time = current_time  # Update last_recorded_time
+
+
+class LoggingMetricCallback(tf.keras.callbacks.Callback):
+  """LoggingMetric callback.
+
+  Log the predefined _PER_BATCH_METRICS after each batch, and log the predefined
+  _PER_EPOCH_METRICS after each epoch.
+  """
+
+  def __init__(self, metric_logger=None):
+    self._logger = metric_logger or logger.BaseBenchmarkLogger()
+    self._per_batch_metrics = _PER_BATCH_METRICS
+    self._per_epoch_metrics = _PER_EPOCH_METRICS
+    self._global_step = 0  # Initialize it in __init__
+    super(LoggingMetricCallback, self).__init__()
+
+  def on_batch_end(self, batch, logs=None):
+    """Log metrics after each batch."""
+    self._global_step += 1
+    for metric in _PER_BATCH_METRICS:
+      self._logger.log_metric(
+          _PER_BATCH_METRICS[metric],
+          logs.get(metric),
+          global_step=self._global_step)
+
+  def on_epoch_end(self, epoch, logs=None):
+    """Log metrics after each epoch."""
+    for metric in _PER_EPOCH_METRICS:
+      self._logger.log_metric(
+          _PER_EPOCH_METRICS[metric],
+          logs.get(metric),
+          global_step=self._global_step)
+
+
+def get_model_callbacks(name_list, **kwargs):
+  """Factory for getting a list of TensorFlow hooks for training by name.
+
+  Args:
+    name_list: a list of strings to name desired callback classes. Allowed:
+      ExamplesPerSecondCallback, LoggingMetricCallback, which are defined
+      as keys in CALLBACKS.
+    **kwargs: a dictionary of arguments to the callbacks.
+
+  Returns:
+    list of instantiated callbacks, ready to be used in a classifier.train call.
+
+  Raises:
+    ValueError: if an unrecognized name is passed.
+  """
+
+  if not name_list:
+    return []
+
+  callbacks = []
+  for name in name_list:
+    callback_name = CALLBACKS.get(name.strip().lower())
+    if callback_name is None:
+      raise ValueError(
+          "Unrecognized training callback requested: {}".format(name))
+    else:
+      callbacks.append(callback_name(**kwargs))
+
+  return callbacks
+
+
+def get_examples_per_second_callback(
+    every_n_steps=1, batch_size=32, metric_logger=None, **kwargs):  # pylint: disable=unused-argument
+  """Function to get ExamplesPerSecondCallback."""
+  return ExamplesPerSecondCallback(
+      batch_size=batch_size, every_n_steps=every_n_steps,
+      metric_logger=metric_logger or logger.get_benchmark_logger())
+
+
+def get_logging_metric_callback(metric_logger=None, **kwargs):  # pylint: disable=unused-argument
+  """Function to get LoggingMetricCallback."""
+  return LoggingMetricCallback(
+      metric_logger=metric_logger or logger.get_benchmark_logger())
+
+
+# A dictionary to map the callback name and its corresponding function
+CALLBACKS = {
```

Comment:
constants should be on the top

Reply:
Yes for common constants. But here its values are two functions defined at the bottom. I think it's more clear to put it here for future extension, just like what we do for hooks.